### PR TITLE
Create CondIsPreferredTool.java

### PR DESCRIPTION
**Pull request by sovdeeth**
### Description

Add a condition to check whether a tool is a preferred tool for a given block or blocks.
Note that preferred tool just means the block will drop an item when broken with this tool, disregarding silk touch mechanics or special cases like shears. For example, a wooden pickaxe is a preferred tool for grass blocks, stone, and leaves, but not for iron ore or obsidian.

---
**Target Minecraft Versions:** 1.16.5+ 
**Requirements:** Paper 1.19.2+ (using blockdatas instead of blocks) 
**Related Issues:** #4185

### CODE REVIEW

File: src/main/java/ch/njol/skript/conditions/CondIsPreferredTool.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+			"\t\tcancel event"
+})
+@Since("INSERT VERSION")
+@RequiredPlugins("1.16.5+, Paper 1.19.2+ (blockdata)")
+public class CondIsPreferredTool extends Condition {
+
+	static {
+		String types = "blocks";
+		if (Skript.methodExists(BlockData.class, "isPreferredTool", ItemStack.class))
+			types = "blocks/blockdatas";
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/conditions/CondIsPreferredTool.java:58]:
```suggestion
			types += "/blockdatas";
```

File: src/main/java/ch/njol/skript/conditions/CondIsPreferredTool.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+		Skript.registerCondition(CondIsPreferredTool.class,
+				"%itemtypes% (is|are) %" + types + "%'s preferred tool",
+				"%itemtypes% (is|are) [the|a] preferred tool (for|of) %" + types + "%",
+			"%itemtypes% (is|are)(n't| not) %" + types + "%'s preferred tool",
+			"%itemtypes% (is|are)(n't| not) [the|a] preferred tool (for|of) %" + types + "%"
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/conditions/CondIsPreferredTool.java:64]:
```suggestion
				"%itemtypes% (is|are) %" + types + "%'s preferred tool[s]",
				"%itemtypes% (is|are) [the|a] preferred tool[s] (for|of) %" + types + "%",
				"%itemtypes% (is|are)(n't| not) %" + types + "%'s preferred tool[s]",
				"%itemtypes% (is|are)(n't| not) [the|a] preferred tool[s] (for|of) %" + types + "%"
```

File: src/main/java/ch/njol/skript/conditions/CondIsPreferredTool.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+		);
+	}
+
+	private Expression<ItemType> items;
+	private Expression<?> blocks;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/conditions/CondIsPreferredTool.java:72]: Import ParseResult directly
```suggestion
	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
```

File: src/main/java/ch/njol/skript/conditions/CondIsPreferredTool.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+				return false;
+			}
+		}
+		// all checks passed
+		return true;
+	}
+
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		return items.toString(event, debug) + " is the preferred tool of " + blocks.toString(event, debug);
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/conditions/CondIsPreferredTool.java:99]: Sounds better \:)
```suggestion
		return items.toString(event, debug) + " is the preferred tool for " + blocks.toString(event, debug);
```

**Review by AyhamAl-Ali — CHANGES_REQUESTED** [review]
Nice PR ⚡

It would be very nice to add a test for this expression as well

**Comment by sovdeeth** [comment]
> It would be very nice to add a test for this expression as well

will do!

File: src/main/java/ch/njol/skript/conditions/CondIsPreferredTool.java (hunk 3 above)
Comment (sovdeeth) [reply src/main/java/ch/njol/skript/conditions/CondIsPreferredTool.java:72]: god damn it every time I think I've finally imported things properly

**Review by AyhamAl-Ali — APPROVED** [review]

**Review by TheLimeGlass — APPROVED** [review]

**Pull request merged**